src/CMakeLists.txt

```diff
@@ -26,8 +26,19 @@ target_sources(libleidenalg
         SurpriseVertexPartition.h)
 set_target_properties(libleidenalg PROPERTIES VERSION ${PROJECT_VERSION})
 set_target_properties(libleidenalg PROPERTIES SOVERSION 1)
-if(WIN32 AND NOT CYGWIN)
-  set_target_properties(libleidenalg PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
-endif()
+
+# Generate export header include file
+include(GenerateExportHeader)
+generate_export_header(libleidenalg
+  STATIC_DEFINE LEIDENALG_STATIC
+  EXPORT_FILE_NAME ${PROJECT_BINARY_DIR}/include/libleidenalg_export.h
+)
+
+# Make sure we can find libleidenalg_export.h
+target_include_directories(
```

Comment:
@theAeon , although this now functions when compiling the library, the `libleidenalg_export.h` header should also still be properly installed. This means that it should be added to the `headers` `FILE_SET`. However, I can't figure out how I would be able to point it towards a different directory (i.e. the `PROJECT_BINARY_DIR` directory). Would you have any idea?

Reply:
It's not the prettiest, but it seems one need to specify the full path. The `BASE_DIRS` setting just strips it.

Reply:
The correct way to do this is actually to revert commit [de44530](https://github.com/vtraag/libleidenalg/pull/1/commits/de4453018c0d1e0238cdd0a1374b2750bc996e20) and add the libleidenalg_export,h to the CMakelists file in `/include` there. Way cleaner.

Reply:
Ha, right, this is more or less what I ended up doing in https://github.com/vtraag/libleidenalg/pull/1/commits/35ee08910c60bc508fa3c49d7d2082ab7628af2c indeed, but I didn't realise it corresponded (almost) to the revert of an earlier commit.